rust/src/delta.rs

```diff
@@ -178,22 +178,28 @@ impl From<StorageError> for LoadCheckpointError {
     }
 }
 
-pub struct DeltaTable {
-    pub version: DeltaDataTypeVersion,
+#[derive(Default)]
+struct DeltaTableState {
     // A remove action should remain in the state of the table as a tombstone until it has expired
     // vacuum operation is responsible for providing the retention threshold
-    pub tombstones: Vec<action::Remove>,
+    tombstones: Vec<action::Remove>,
+    files: Vec<String>,
+    commit_infos: Vec<Value>,
+    app_transaction_version: HashMap<String, DeltaDataTypeVersion>,
+}
+
+pub struct DeltaTable {
+    pub version: DeltaDataTypeVersion,
     pub min_reader_version: i32,
     pub min_writer_version: i32,
     pub table_path: String,
 
+    state: DeltaTableState,
+
     // metadata
     // application_transactions
     storage: Box<dyn StorageBackend>,
 
-    files: Vec<String>,
-    app_transaction_version: HashMap<String, DeltaDataTypeVersion>,
-    commit_infos: Vec<Value>,
     current_metadata: Option<DeltaTableMetaData>,
     last_check_point: Option<CheckPoint>,
```

Comment:
i think field being mutated in `process_action` method should go into the state struct. then we can probably just pass in a mut reference of state into `process_action` instead of self.

Reply:
done. Looks a bit clearer now, separating the meta data (path, backend, ...) from the content.

Had to introduce getters for `min_reader_version` and `min_writer_version`, which looks a bit odd.

Anything else? Squashing?

Reply:
For future reference, you can leave the incremental commits in the PR if you prefer, i will just do a squash merge at the end.